### ISSUE TITLE
active-repositories in cabal.project does not override cabal.config

### ISSUE THREAD

**Issue by brandonchinn178** [haskell/cabal#9081]
**Describe the bug**
Running `ghc-wasm-meta`'s `setup.sh` script adds `active-repositories: hackage.haskell.org, head.hackage.haskell.org:override` to `~/.cabal/cabal.config` and runs `cabal update`.

In my project, I have `cabal.project` with `active-repositories: hackage.haskell.org`, but `cabal update` is still showing head.hackage. I would expect `active-repositories` in `cabal.project` to override `cabal.config`; is that assumption wrong? How do I prevent using head.hackage without patching the script to not set head.hackage?

**System information**
 - Operating system
 - `cabal`, `ghc` versions

**Additional context**

**Comment by ulysses4ever** [comment]
This is closely related to 

- #8730

tldr; is that overriding is not currently implemented, and it's not clear that it should override one another: you could argue either way. There is a more general design question: how to make all users happy, those who want override-semantics and those who want add-semantics, so that we avoid doing an arbitrary irreversible choice between the semantics. The way forward would be to have someone proposing a design supporting both semantics, and if people are happy with it, implementing it.

**Comment by andreabedini** [comment]
Other related tickets are #8997, #8863, and #8502
